docs/metadata.rst

```diff
@@ -59,66 +63,190 @@ JSON
 When ``json`` is specified as the ``codec`` in the schema the metadata is encoded in
 the human readable `JSON <https://www.json.org/json-en.html>`_ format. As this format
 is human readable and encodes numbers as text it uses more bytes than the ``struct``
-format. However it is simpler to configure as it doesn't require any format specifiers
-in the schema.
+format. However it is simpler to configure as it doesn't have any specific constraints and
+doesn't require any format specifiers for each type in the schema.
 
 ------
 struct
 ------
 
 When ``struct`` is specifed as the ``codec`` in the schema the metadata is encoded
-using :py:func:`struct.pack` which results in a compact binary representation. This
-codec places extra restrictions on the schema:
-
-#. Each property must have a ``binaryFormat`` entry.
-    This describes the encoding for that property using ``struct``
-    `format characters <https://docs.python.org/3/library/struct.html#format-characters>`_.
-    For example an unsigned 8-byte integer can be specifed with::
-
-        {'type': 'number', 'binaryFormat':'Q'}
-
-    And a length 10 string with::
-
-        {'type': 'string', 'binaryFormat':'10p'}
-
-    * Endian-ness is fixed at little endian so cannot be specified.
-    * Not all formats are supported - ``binaryFormat`` must match
-      ``^[cbB\?hHiIlLqQfd]|\d*[spx]$``.
-    * It is recommended to use ``p`` for strings, as ``s`` doesn't store where
-      the string ends. Note that ``5p`` can only store a string that when encoded is 4
-      bytes or less. If using ``s``, set ``nullTerminated`` to ``true`` if you are
-      sure your strings do not have null characters in. This will remove the spare
-      bytes at the end of the string.
-    * Strings that are longer than the specified length will be silently truncated,
-      note that the length is in bytes, not characters.
-    * To insert padding bytes use ``null`` for ``type`` and ``x`` for ``binaryFormat``.
-    * For ``array`` types the format of the length of the array can be chosen with
-      ``arrayLengthFormat`` which must be one of ``B``, ``H``, ``I``, ``L`` or ``Q``
-      which is the default. For dealing with legacy encodings that do not store the
-      length of the array setting ``noLengthEncodingExhaustBuffer`` to true will read
-      the buffer to exhaustion. As such an array with this encoding must be the last type
-      in the encoded struct.
-    * For strings, the encoding can be set with ``stringEncoding`` which defaults to
-      ``utf-8``.
+using :py:func:`struct.pack` which results in a compact binary representation which is much smaller
+and generally faster to encode/decode than JSON.
+
+This codec places extra restrictions on the schema:
+
+#. Each property must have a ``binaryFormat``
+    This sets the binary encoding for used for the property.
```

Comment:
"for used for"

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/741/commits/0b31382e4eb6dc7aec6a665388ff1f2a8cd8b548